lib/import/import-manager.ts

```diff
@@ -1,12 +1,42 @@
-import { ContentItemModels, LanguageVariantModels, ManagementClient } from '@kontent-ai/management-sdk';
-
-import { defaultExternalIdGenerator, getDefaultLogger, getMigrationManagementClient, Logger } from '../core/index.js';
+import { ManagementClient } from '@kontent-ai/management-sdk';
+import chalk from 'chalk';
+import { defaultExternalIdGenerator, extractErrorData, getDefaultLogger, getMigrationManagementClient, Logger } from '../core/index.js';
 import { importContextFetcherAsync } from './context/import-context-fetcher.js';
-import { ImportConfig, ImportContext, ImportResult } from './import.models.js';
+import { ImportConfig, ImportContext, ImportedItem, ImportedLanguageVariant, ImportResult } from './import.models.js';
 import { assetsImporter } from './importers/assets-importer.js';
 import { contentItemsImporter } from './importers/content-items-importer.js';
 import { languageVariantImporter } from './importers/language-variant-importer.js';
 
+const reportFilename: string = `import-report.json`;
+
+type ReportResult = {
```

Comment:
It may be ideal to look into a `DeepReadonly` utility type.
Something like the following: [https://stackoverflow.com/questions/41879327/deepreadonly-object-typescript](https://stackoverflow.com/questions/41879327/deepreadonly-object-typescript)

Reply:
Even though it is a nice type function I don't think that's necessary :)
The type is not that nested it would need this :D